FIR Completion: offer Java classes

tgeng opened the pull request:
This PR needs https://kotlin.jetbrains.space/p/kotlin/reviews/247/timeline

darthorimar commented:
There was some compilation problems in this PR which I fixed myself.

But this PR also breaks some FIR completion tests. Could you please take a look at them?

```
HighLevelJvmBasicCompletionTestGenerated$Common.testExtensionFunReceiver 
HighLevelJvmBasicCompletionTestGenerated$Common.testInExpressionNoPrefix 
HighLevelJvmBasicCompletionTestGenerated$Common.testFromImports 
HighLevelJvmBasicCompletionTestGenerated$Common.testInExtendTypeAnnotation 
HighLevelJvmBasicCompletionTestGenerated$Common$Annotations.testTopLevelAnnotationArg1 
HighLevelJvmBasicCompletionTestGenerated$Common$Annotations.testOnlyAnnotations 
HighLevelJvmBasicCompletionTestGenerated$Java.testNonImportedPrefixMatching1 
HighLevelJvmBasicCompletionTestGenerated$Java.testNonImportedPrefixMatching2 
HighLevelJvmBasicCompletionTestGenerated$Java.testJavaClassNames 
HighLevelMultiFileJvmBasicCompletionTestGenerated.testNotImportedJavaClass 
HighLevelBasicCompletionHandlerTestGenerated.testJavaSAM 
```

tgeng commented:
> There was some compilation problems in this PR which I fixed myself.
> 
> But this PR also breaks some FIR completion tests. Could you please take a look at them?
> 
> ```
> HighLevelJvmBasicCompletionTestGenerated$Common.testExtensionFunReceiver 
> HighLevelJvmBasicCompletionTestGenerated$Common.testInExpressionNoPrefix 
> HighLevelJvmBasicCompletionTestGenerated$Common.testFromImports 
> HighLevelJvmBasicCompletionTestGenerated$Common.testInExtendTypeAnnotation 
> HighLevelJvmBasicCompletionTestGenerated$Common$Annotations.testTopLevelAnnotationArg1 
> HighLevelJvmBasicCompletionTestGenerated$Common$Annotations.testOnlyAnnotations 
> HighLevelJvmBasicCompletionTestGenerated$Java.testNonImportedPrefixMatching1 
> HighLevelJvmBasicCompletionTestGenerated$Java.testNonImportedPrefixMatching2 
> HighLevelJvmBasicCompletionTestGenerated$Java.testJavaClassNames 
> HighLevelMultiFileJvmBasicCompletionTestGenerated.testNotImportedJavaClass 
> HighLevelBasicCompletionHandlerTestGenerated.testJavaSAM 
> ```

Weird... Not sure what I did that causes some code to be deleted when rebasing and this causes the test to fail. Anyway, I just fixed that and it should work now.

darthorimar commented:
Now everything is ok :) merged, thank you!
